### PR TITLE
Hide ImageTextAlternative balloon when image element has been removed

### LINKED ISSUE #137: Image balloon is not removed together with image on external changes
sep-08-2017 16-58-19

### LINKED ISSUE #5115: Image balloon is not removed together with image on external changes
sep-08-2017 16-58-19

### PR DESCRIPTION
**Pull request by oskarwrobel**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Fix: Hide `ImageTextAlternative` balloon when image element has been removed by external changes. Closes ckeditor/ckeditor5#5115.

### CODE REVIEW

File: tests/image/utils.js [hunk 1]
```diff
@@ -49,6 +52,40 @@ describe( 'image widget utils', () => {
...
+
+			const selection = new ViewSelection( [ ViewRange.createOn( notWidgetizedElement ) ] );
+
+			expect( isImageWidgetSelected( selection ) ).to.be.false;
+		} );
+
+		it( 'should return false when widget element is not the only element in the selection', () => {
+			const notWidgetizedElement = new ViewElement( 'p' );
+
+			frag = new ViewDocumentFragment( [ notWidgetizedElement, notWidgetizedElement ] );
```
Comment (oleq) [inline tests/image/utils.js]: Shouldn't this be `new ViewDocumentFragment( [ element, notWidgetizedElement ] )` instead?

**Comment by oleq** [comment]
Shouldn't this issue be included in http://localhost:8125/ckeditor5-ui/tests/manual/contextualballoon/externalchanges.html or provide a new MT?

**Review by oleq — CHANGES_REQUESTED** [review]
Minor changes needed in tests.

**Comment by oskarwrobel** [comment]
> Shouldn't this issue be included in http://localhost:8125/ckeditor5-ui/tests/manual/contextualballoon/externalchanges.html or provide a new MT?

I was thinking about it but this is something that we are able to test by our unit tests. I think MT is useful to test repositioning because this is something we need to stub in unit tests.

But I can add MT if you think is needed for this case.

**Comment by oleq** [comment]
For one thing, I was surprised that the MT is in ckeditor5-ui instead of ckeditor5-link. The code which handles re-positioning and stuff on `#render` isn't in the `ContextualBalloon`. It's a pure link stuff. 

Then I thought that we should check the same thing in the image and I finally found out there's a corresponding test in the image too ckeditor5-image/tests/manual/tickets/106/1.html. It doesn't test this kind of issue because what is "remotely" removed is the paragraph before the image instead of the image widget itself.

So maybe, let's just extend this MT and include this issue's case?

**Comment by oskarwrobel** [comment]
I was using `106` MT to prepare the GIF for this issue. OK, I'll extend this test.

**Comment by oleq** [comment]
The MT is OK. But it revealed that the editor loses focus when the image which text alternative is being edited gets removed. This, I'm afraid, must be addressed (it [works](https://github.com/ckeditor/ckeditor5-link/blob/master/src/link.js#L246-L248) in the link feature because it explicitly calls focus).

**Pull request merged**